wire-schema/src/main/java/com/squareup/wire/schema/Pruner.kt

```diff
@@ -90,16 +90,29 @@ internal class Pruner(
 
     // The top-level type isn't a root, search for root members inside.
     for (reachable in reachableObjects(protoType)) {
-      if (reachable is ProtoMember) {
-        if (identifierSet.includes(reachable)) {
-          marks.root(reachable)
-          marks.mark(reachable.type) // Consider this type as visited.
-          queue.add(reachable)
-        }
+      if (reachable !is ProtoMember) continue
+      if (!isRetainedVersion(reachable)) continue
+      if (identifierSet.includes(reachable)) {
+        marks.root(reachable)
+        marks.mark(reachable.type) // Consider this type as visited.
+        queue.add(reachable)
       }
     }
   }
 
+  /** Returns true if this member survives `since` and `until` pruning. */
+  private fun isRetainedVersion(protoMember: ProtoMember): Boolean {
```

Comment:
nit: extension function?

Reply:
I don’t like extension functions on things that rely on state that those things don’t have? In this case ProtoMember is an identifier, but this method needs to use the schema to make the decision.